### ISSUE TITLE
Implement UX for users to add a donation to Giveth for every donation

### ISSUE THREAD

**Issue by divine-comedian** [Giveth/giveth-dapps-v2#1579]
Following conclusive GIVeconomy research and forum discussion on revenue streams we should begin thinking of how to ask users to give a % of the donation amount to Giveth DAO - these extra donation amounts will be used to fund Giveth's day to day operations.

Here's an shot of the modal from the Gitcoin Grants UI we can look to as an example:

![image.png](https://images.zenhubusercontent.com/5fff1c86d883b3d9a858ddbc/14650101-8193-4aff-9125-447fc5a1d4ac)

The idea here is that we automatically add a % fee on the donation amount, the user can set the fee to 0% (turn it off) or set another amount. 

We should make it very easy, almost thoughtless for users to make donations and give a percentage to Giveth.

We should make it mildly inconvenient, but possible for users to change the amount to 0% 

Looping @laurenluz and @GriffGreen into the conversation for any points I might have missed.

**Comment by markoprljic** [comment]
Easiest and simplest way to add this in the UI is adding this little piece on the Donation page, and include the cost breakdown on that page and in the donation confirmation modal as well. 

<img width="262" alt="Screenshot 2022-09-29 at 09 45 34" src="https://user-images.githubusercontent.com/5704285/192971399-dc1e3533-fbb0-48ab-9926-bca38d9e0b53.png">

From design side this can be done in few hours. Dev need to estimate their effort.

@mosaeedi @MoeNick

**Comment by MoeNick** [comment]
Does it approved in the forum to add fees?

**Comment by divine-comedian** [comment]
> Does it approved in the forum to add fees?

yes

**Comment by mosaeedi** [comment]
![Screen Shot 2022-10-11 at 8 03 18 PM](https://user-images.githubusercontent.com/13708322/195154977-b1016e92-d952-40e7-8dcf-3a962b1dfdf0.png)
I had to redo the whole screen to make it look good!
As you can see, I've added two sections, 

## First-> Donate to Giveth

![Screen Shot 2022-10-11 at 8 04 40 PM](https://user-images.githubusercontent.com/13708322/195155197-9eadf03b-a713-47bf-9430-29dcf9aa5876.png)

## Second -> Show the total donation to the project and Giveth

![Screen Shot 2022-10-11 at 8 05 26 PM](https://user-images.githubusercontent.com/13708322/195155342-862d128f-b63c-4450-9fe5-7e0231b49e54.png)


I made some changes too, like moving the **Make it anonymous** to the bottom.

What do you guys think? @markoprljic @Tosinolawale @MoeNick @laurenluz @divine-comedian 

If you need to take a closer look -> [Figma link](https://www.figma.com/file/nVoinu0tgJ565enN5R4WDE/Giveth.io-%26-GIVeconomy?node-id=14138%3A253052) [Go to your right **->** on Figma]

**Comment by markoprljic** [comment]
Looks good!

**Comment by divine-comedian** [comment]
just a small comment @mosaeedi how would it look in the text box to keep the `%` symbol outside the custom input box, just to the left and make the custom % input box itself smaller.

we could also remove the 0% option or replace it with 20%. 

If the user wants to select 0% it could be a check box

- [x] I do not want to support Giveth with part of my donation.

just another idea to make it more inconvenient to remove the default donation

**Comment by markoprljic** [comment]
Love inconvenience suggestions by Mitch. 😃

**Comment by MoeNick** [comment]
@aminlatifi you can provide your ideas here.

**Comment by aminlatifi** [comment]
> @aminlatifi you can provide your ideas here.

It's totally a frontend task and can be implemented shortly if the design be finalized.

**Comment by divine-comedian** [comment]
Also just a small detail - should we consider taking a % of the amount the donor specifies to donate or add a percentage on top? 
i.e 
donates 900 GIV to a project + 10% to Giveth

should that equal 900 GIV to the project and 90 to Giveth 
OR
810 to the project and 90 to Giveth

the former stays in line with the slogan "100% of your donation goes to the project" while the logic becomes confusing when a donor would like to simply donate all of their tokens to a project - we can't add an amount on top if they are already donating the max.

I would be in favour of the latter option
> 810 to the project and 90 to Giveth

**Comment by markoprljic** [comment]
I'd add it on top. It's a common sense when selecting % on already selected
amount, it's expected results to see amount + % selected.

**Comment by divine-comedian** [comment]
Another note to consider is the edge case where the project being donated to is Giveth itself. 

In this case I think it makes sense that we wouldn't split the donation and perhaps hide the elements relating to that functionality

or if there's something else we should show the user in this very specific case.

**Comment by markoprljic** [comment]
Great catch @divine-comedian 

I'd leave the effort estimate to the devs but if it turns out to be large, then IMO I wouldn't make any significant difference in the UI for the edge case, to avoid complicating things both on FE And BE. 
We can just default the donation to 0%. 

Up to devs to estimate and let us know.

**Comment by MoeNick** [comment]
- [ ] redesign the modals - having steps in modal
- [ ] what to do with "you are a giver" now page/
- [ ] if the modal gets closed,  what to in every steps? 

Questions: 
1- for some third-party integrations (like giving-block and CHANGE ) some of them differ in the list of tokens and the supported networks, also some of the projects don't accept funds in specific networks. So, what if a XXX token is acceptable on the project  for the current user network and not acceptable for GIVETH, I know it's not a case now because we accept all tokens in all supported networks, but for future H-Cases we have to make sure the token and the network is ok for these 2 projects. 

2- Do we have to apply this handler for all projects? including third party projects? 

3- Do we want to have different Giveth projects for the second trx in the future?

**Comment by MoeNick** [comment]
4- input % field limit: 0-%100? why not more than %100 if you want to add it on top?

**Comment by divine-comedian** [comment]
> * [ ]  redesign the modals - having steps in modal
> 
>     * [ ]  what to do with the "you are a giver now" page.
> 
>     * [ ]  if the modal gets closed,  what to in every step?
> 
> 
> Questions: 1- for some third-party integrations (like giving-block and CHANGE ) some of them differ in the list of tokens and the supported networks, also some of the projects don't accept funds in specific networks. So, what if a XXX token is acceptable on the project for the current user network and not acceptable for GIVETH, I know it's not a case now because we accept all tokens in all supported networks, but for future edge-Cases we have to make sure the token and the network is ok for these 2 projects.
> 
> 2- Do we have to apply this handler for all projects? including third party projects?
> 
> 3- Do we want to have different Giveth projects for the second trx in the future?

1- I think this is an edge case we'll very likely never have to worry about. Giveth will always have a project address deployed on any network that the dapp uses and we won't restrict the tokens that can be accepted.

2 - If by third party projects like giving block then I would say yes

3- If we have a cart of projects feature in the future perhaps we would ask the user if they wanted to donate to the matching pool - but this would likely be very far in the future. The main focus at the moment is creating revenue for Giveth to allow the DAO to sustain it's costs.

**Comment by RamRamez** [comment]
What is the content of the tooltip?
![image](https://user-images.githubusercontent.com/73441857/196543471-9502d870-73c7-41f0-b56e-118e1174cf38.png)

**Comment by divine-comedian** [comment]
> What is the content of the tooltip? ![image](https://user-images.githubusercontent.com/73441857/196543471-9502d870-73c7-41f0-b56e-118e1174cf38.png)

"Support Giveth to continue building the Future of Giving by making a donation to the Giveth DAO! The percentage you choose will be added on top of your donation amount, in the same token you are donating."

**Comment by divine-comedian** [comment]
this modal here could be made a bit bigger - the top part showing the numbers for the 2 donation parts is a bit too small 

![image](https://user-images.githubusercontent.com/67759413/197038692-e7dfbc90-f0d2-4175-8d08-0e97828e7057.png)

**Comment by laurenluz** [comment]
@divine-comedian @RamRamez I think that donate to Giveth should be the 2nd tx because it was an "add-on"... this order feels weird to me

open to you disagreeing.

**Comment by divine-comedian** [comment]
> @divine-comedian @RamRamez I think that donate to Giveth should be the 2nd tx because it was an "add-on"... this order feels weird to me
> 
> open to you disagreeing.

I think the logic was that if the user rejects the second transaction or leaves the view or loses the modal at least we get the donation to giveth in the first transaction.

**Comment by laurenluz** [comment]
> I think the logic was that if the user rejects the second transaction or leaves the view or loses the modal at least we get the donation to giveth in the first transaction, which they are least likely to mess up.

It just feels very big brother to me: "you must donate to Giveth before you can donate to the project that brought you here"

(if this comment doesn't sway you I will just stop talking and let you do it how you like).

**Comment by markoprljic** [comment]
Project first, Giveth second.

**Comment by Tosinolawale** [comment]
> Hey @laurenluz @divine-comedian @markoprljic We need a single point of truth for this issue, we were working on a flow with @mosaeedi which is now blocked by his vacation, so I can't help until he's back and the whole flow and the design get ready and then we jump into developing. If anyone else wants to do the PMing of this issue, (which I suggest @divine-comedian wear the hat) please define it well so we can include all backed and front-end and I can analyze it for implementation.
> 
> So for now, I told @RamRamez to stop it until we have a straightforward design.

Hi @MoeNick, I'll take a look at this and take over

**Comment by markoprljic** [comment]
> If the user wants to select 0% it could be a check box
> 
> * [x]  I do not want to support Giveth with part of my donation.

No checkbox needed here, there is option for 0% already. 

If user wants more than 100% they can enter it in the field. 
% sign should stay inside the input field.

**Comment by markoprljic** [comment]
I'm noticing a lot of discussion for a simple enhancement that should be straightforward to design and implement, more so because it is almost a common pattern seen elsewhere (example Gitcoin). We shouldn't reinvent the wheel or try to add more options for the user that could hinder them from completing donation. It's a simple choice, a "click" away, and it should stay so. 

[Current Figma version](https://www.figma.com/file/nVoinu0tgJ565enN5R4WDE/Giveth.io-%26-GIVeconomy?node-id=14138%3A253052) looks complete to me. @MoeNick 

Deployed version should not have a limitation for user to enter custom % amount. https://default-donations.vercel.app/donate/how-many-photos-is-too-many-photos

Donation modal, 2tx process, since it is re-used throughout the app it should stay as is for now, unless we want to redesign it and apply everywhere else. This should then be another issue treated separately from this one.

**Comment by MoeNick** [comment]
The redesign of modal and flow of showing success page is matter @markop

**Comment by divine-comedian** [comment]
I'm cool with Marko's ideas - might me another case of too many cooks

@MoeNick I can see Ramin finished the 2tx modal - it looks mostly fine. I think maybe a simple revision of the success page would be the last step. Unless I'm missing something I don't think this issue really is that complicated.

**Comment by MoeNick** [comment]
If user close the modal, shall we open it up for next trx? Success means success for both? If user rejects the second one, sholud't we allow him to know the first one went ok? This is the user flow after I mean

**Comment by markoprljic** [comment]
If user closes the modal they will still have pending notification in their Web Wallet (Metamask or other). Later when we will have our Notification center it will also show pending tx's there. 

I believe it should behave the same way it does on the GIVfarm for Approve/Stake two tx's. At any point user can reject or close the modal (while still having Pending tx in metamask). If continued through Metamask it will auto-trigger second tx, after which we show Success message. 

While we're at this topic, I wonder what Gitcoin is using to bundle everything into a single tx and disperse the donations to projects?

**Comment by divine-comedian** [comment]
> While we're at this topic, I wonder what Gitcoin is using to bundle everything into a single tx and disperse the donations to projects?

Actively working on this in GIVfi research :wink:

**Comment by markoprljic** [comment]
Hooray for the research!

**Comment by divine-comedian** [comment]
- What to do if user rejects second tx - do we take them to success page? 
- What to do if user rejects first tx? 
- Is the success page different?

**Comment by markoprljic** [comment]
Knowing that the second tx will show pending in web wallet, and user can decide to reject it (unlikely), if they do so, they have done it intentionally and we can assume they expect to see confirmation of the first transaction. In that context ...

> What to do if user rejects second tx - do we take them to success page?
Yes. Default success donation of the selected project.

> What to do if user rejects first tx?
Show modal tx rejected, abort the whole process.

**Comment by divine-comedian** [comment]
> Yes. Default success donation of the selected project.

Do we also show the tx rejected modal before taking them to the success page?

**Comment by markoprljic** [comment]
Yes, because the system should always provide feedback on user actions.

**Comment by Tosinolawale** [comment]
> testing on: https://default-donations.vercel.app/donate/how-many-photos-is-too-many-photos browser: firefox OS: ubuntu 22.04
> 
> @RamRamez I'm just going to dump feedback as i come across
> 
> make the tooltip text just a few points smaller - it takes up a lot of the page ![image](https://user-images.githubusercontent.com/67759413/197029040-ddbf0c8d-3898-47ec-bcb3-f57487b6571c.png)
> 
> Can we also implement this change here?
> 
> > just a small comment @mosaeedi how would it look in the text box to keep the `%` symbol outside the custom input box, just to the right and make the custom % input box itself smaller.
> > we could also remove the 0% option or replace it with 20%.
> > If the user wants to select 0% it could be a check box
> > ```
> > * [x]  I do not want to support Giveth with part of my donation.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > just another idea to make it more inconvenient to remove the default donation
> > If you're crunched for space on the card I would recommend removing the cryptocurrency / credit card toggle at the top up - tbh it's not coming soon and might not be coming at all
> 
> Maybe something like this? in lieu of @mosaeedi I tried my hand - maybe @markoprljic can tell me if it's okay If left unchecked user can set any percent they want. ![image](https://user-images.githubusercontent.com/67759413/197035019-9f2f2e2e-6be5-4ba7-b4f5-b4015cb0568b.png)
> 
> If the user checks the box the donation amount is set to 0% If the user sets donation to 0% then the box becomes checked ![image](https://user-images.githubusercontent.com/67759413/197035442-1fe44bbe-a6c2-4900-a0e9-204165ef283d.png)
> 
> If the user sets an amount after then this box becomes unchecked again.
![Wallet   is    connected](https://user-images.githubusercontent.com/35389241/197612667-65e59dd9-a77d-4f0f-81a4-d7ca52a96dba.png)
For starters, we can use this tool tip, we should be consistent with the tool tip we use across board. I also made changes based on Mitch's suggestion on the input modals

**Comment by divine-comedian** [comment]
> ![Wallet is connected](https://user-images.githubusercontent.com/35389241/197617619-8688b4e9-b779-4282-b32a-98d615feed3d.png)
> 
> For starters, we can use this tool tip, we should be consistent with the tool tip we use across board. I also made changes based on Mitch's suggestion on the input modals

Thanks @Tosinolawale ! Just a few small grammar things for the tooltip, should read:
"Support Giveth with a donation to the Giveth DAO. The selected donation percentage will be added on top of your donation amount in the same token you are donating."

--- 

Last issue I think is this page - @Tosinolawale 
![image](https://user-images.githubusercontent.com/67759413/197622032-4ce0c587-131a-47b0-ba05-27561be9c8ff.png)

- Should we have some additional content for when a user donates to Giveth as well? 
> i.e Thanks for donating to . . . 
> We're also grateful for your donation to the Giveth DAO, you can check out our [project on Giveth here](https://giveth.io/project/the-giveth-community-of-makers).
- Should we provide the user a link to both transactions they just made?

**Comment by Tosinolawale** [comment]
![Group 623](https://user-images.githubusercontent.com/35389241/197653881-b4d97e15-f696-4a63-97ee-dd1ad49e2be7.png)
@divine-comedian

**Comment by Tosinolawale** [comment]
![Group 624](https://user-images.githubusercontent.com/35389241/197654255-dbcfa17e-28d5-4cd5-82cb-9fc1d277e00c.png)
@markoprljic @divine-comedian also drafted the donation flow to this

**Comment by divine-comedian** [comment]
Hey @Tosinolawale thanks for the update but there's a few things that don't look quite right
the first two modals you show should look similar to this:
![image](https://user-images.githubusercontent.com/67759413/197670415-54ba5a37-6ca6-4a6a-a1fe-df226421746e.png)
This is because it is just a peer to peer transaction, not a smart contract interaction, there is no 'approve' and no spend limit.

For the cases you mention of close/canceled this is actually two separate cases
- user closes modal
- user cancels web wallet tx
What happens in each? 
If the user closes the modal after finishing the first tx or accidentally clicks away does that mean the donation to giveth fails? they would still have the pending tx in their web wallet...
If they actually cancel the tx on their web wallet this would normally throw an error message...
---

In the modal below if the user cancels the tx on their web wallet then there is no tx to view on etherscan - the associated text should be omitted in that particular case. If the error fails for another reason then there might be a failed tx to view @RamRamez would know better...
![image](https://user-images.githubusercontent.com/67759413/197669995-e62ac4ce-a872-402e-96bb-ec0ae550ee9a.png)


Also some smaller copy fixes - I'll follow up with comments in figma

**Comment by divine-comedian** [comment]
Okay @RamRamez 

I think it's important to get this issue rolling and close it out. I'll try to summarize the changes as best as possible.

- [ ] revert order of transactions - project donation first, giveth donation second
- [ ] Implement modal for two transactions https://www.figma.com/file/nVoinu0tgJ565enN5R4WDE/Giveth.io-%26-GIVeconomy?node-id=14511%3A263642
- [ ] If the user rejects the first transaction on their web wallet then we do not generate the second one automatically - we abort the donation process - but we still show them an error
- [ ] Modal reacts to donation states - If each donation is sucessful or if one donation fails - here is the modal states https://www.figma.com/file/nVoinu0tgJ565enN5R4WDE/Giveth.io-%26-GIVeconomy?node-id=14511%3A264013
- [ ] when both donations are successful we reflect it on the modal briefly then take the user to the success page
- [ ] if the second transaction fails we reflect it on the modal briefly and take the user to the success page
- [ ] the copy on the success page is different if the user successfully made a giveth donation and we provide a link to the second (giveth) donation - copy should read: 

> Thank you for supporting The Giveth Community of Makers and thanks for your donation to the Giveth DAO! You can check out the Giveth DAO project here(ADD LINK TO https://giveth.io/project/the-giveth-community-of-makers).

- [ ] tool tip on project donate page should read:
> Support Giveth with a donation to the Giveth DAO. The selected donation percentage will be added on top of your donation amount in the same token you are donating.

I will probably come behind on your branch and make many text fixes.

does most of this make sense?

**Comment by RamRamez** [comment]
> 

Thanks you @divine-comedian for this great summarization. It really saved my time. Some other points:
1. Adding `I do not want to support Giveth with my donation` checkbox and `20%` button
2. Remove Input limitation (current limitation is 100%. Users can't input more than that)
3. Modal shouldn't be closed when clicking outside modal.

Do you confirm these to be added?

**Comment by divine-comedian** [comment]
> Thank you @divine-comedian for this great summarization. It really saved my time. Some other points:
> 
>     1. Adding `I do not want to support Giveth with my donation` checkbox and `20%` button
> 
>     2. Remove Input limitation (current limitation is 100%. Users can't input more than that)
> 
>     3. Modal shouldn't be closed when clicking outside modal.
> 
> 
> Do you confirm these to be added?

Yes to points 1 & 2

3. Talking with @Tosinolawale I think we concluded that the modal can be closed when the user clicks outside -so no change in current behaviour

**Comment by divine-comedian** [comment]
@RamRamez going through some testing - one small change I would make is in the state between user completes the first transaction and has not started the second:

![image](https://user-images.githubusercontent.com/67759413/199097601-ba647012-fcc2-4916-ab39-900806719fa1.png)

- [ ] In the modal the text for the second transaction should say "You are donating" instead of "Donation submitted" and should change when the donation is submitted or when the tx is actually confirmed and sent
- [ ] the size of the checkbox on "I do not want to support . . ." is very small - it's hard to see the actual check - is it possible to make the checkbox bigger but keep the text the same size?

![image](https://user-images.githubusercontent.com/67759413/199099076-49d2df66-41cf-4511-bd59-7b88315114b8.png)


other small fixes within my skill I made here:
https://github.com/Giveth/giveth-dapps-v2/commit/ad794255acaeb8646905721361dd6d092acbfe38

**Comment by maryjaf** [comment]
Please check this scenario:
I choose the "How many photos is too many photos?" project for donation
I set  0.000003 GIV and select  20%  Donate to Giveth
After choosing donate option and confirm the first transaction, I get this error
![image.png](https://images.zenhubusercontent.com/62fe7a8b57c1a1770575ea40/721c50d6-fc8a-427a-b7ff-dcf0c2426c4b)
And then the second picture
![image.png](https://images.zenhubusercontent.com/62fe7a8b57c1a1770575ea40/54671ad2-b423-414e-950b-ac3810207d4d)
And in fact, the first transaction is done and displayed in the list of project donations
and the second transaction is fielded
@RamRamez

**Comment by RamRamez** [comment]
> @RamRamez going through some testing - one small change I would make is in the state between user completes the first transaction and has not started the second:
> 
> ![image](https://user-images.githubusercontent.com/67759413/199097601-ba647012-fcc2-4916-ab39-900806719fa1.png)
> 
> * [x]  In the modal the text for the second transaction should say "You are donating" instead of "Donation submitted" and should change when the donation is submitted or when the tx is actually confirmed and sent
> * [x]  the size of the checkbox on "I do not want to support . . ." is very small - it's hard to see the actual check - is it possible to make the checkbox bigger but keep the text the same size?
> 
> ![image](https://user-images.githubusercontent.com/67759413/199099076-49d2df66-41cf-4511-bd59-7b88315114b8.png)
> 
> other small fixes within my skill I made here: [ad79425](https://github.com/Giveth/giveth-dapps-v2/commit/ad794255acaeb8646905721361dd6d092acbfe38)
> 
> other than that everything seemed to work great!

I fixed these two points mentioned here.
Beside that I disabled closing modal on click outside (for donate modal only)

**Comment by RamRamez** [comment]
> Please check this scenario: I choose the "How many photos is too many photos?" project for donation I set 0.000003 GIV and select 20% Donate to Giveth After choosing donate option and confirm the first transaction, I get this error ![image.png](https://camo.githubusercontent.com/1e79a1c8013b3d9692478eba30570e831d2ec1bd48b29a4a2474bd39a7ded1bf/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3632666537613862353763316131373730353735656134302f37323163353064362d666338612d343237612d623766662d646366306332343236633462) And then the second picture
> 
> ![image.png](https://camo.githubusercontent.com/edc3a73833fc85cdace270db1d92a8824835b15f4254a1e6eac476ad94ca2e0e/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3632666537613862353763316131373730353735656134302f35303937356133652d313038312d343338342d396137662d613433656336326231653363) In fact, the first transaction is done and displayed in the list of project donations and the second transaction is fielded @RamRamez

Thanks @maryjaf, nice catch.
Actually we don't care about second tx. No matter if it's failed, rejected or not saved on our DB. We show a toast error on top the screen and after 4 seconds success view is shown.
So in your scenario, second tx has failed and then success view is shown and it's ok.

But in toast error, the problem occurs for very low donation amounts. I fixed it by setting a minimum.
The minimum donation amount had been 0.000001.
So, I set this for donation to Giveth too.
It means if user chooses to donate 0.000001 GIV to a project and chooses 1% to donate to Giveth, both txs will be 0.000001

@divine-comedian Do you confirm this?

**Comment by maryjaf** [comment]
I think that if any transaction was unsuccessful, it is better to record the result of each transaction in the view.

**Comment by RamRamez** [comment]
> I think that if any transaction was unsuccessful, it is better to record the result of each transaction in the view.
> 
> Is it possible that the first transaction is unsuccessful and the second is successful? Or if the first one fails, the process will stop

We show the result of both transactions in donate modal. But in success view, we just show the tx hashes. (If user reject the tx, there's no tx hash)

Yeah, if first one fails for any reason, second one won't be started

**Comment by divine-comedian** [comment]
testing on https://default-donations.vercel.app/
check box size looks good
cannot close the modal accidentally. :)
donation text changes when second tx is submitted

I pushed a small text improvement to the error message when there is too many decimal places

![image](https://user-images.githubusercontent.com/67759413/199282576-893edeae-61de-4dd4-be6a-12407862c3ff.png)


I think everything looks great!

**Comment by maryjaf** [comment]
![image.png](https://images.zenhubusercontent.com/62fe7a8b57c1a1770575ea40/875152d5-a684-4f58-b59e-6ae1f00456cd)
@RamRamez plz check this

**Comment by RamRamez** [comment]
> ![image.png](https://camo.githubusercontent.com/e3aeefeb30a002a489a0eb19376edb46d9d51a7d6a9e82f6d09f83e6b76aef2d/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3632666537613862353763316131373730353735656134302f38373531353264352d613638342d346635382d623539652d366165316630303435366364) @RamRamez plz check this In fact, it is only a display problem and we do not show 0 GIV, but when we tap on DONATE , only one transaction is shown

I fixed it. Please check again.

**Comment by GriffGreen** [comment]
Wow you are working late Ramin! 

I went to test but can't seem to get the vercel page to load

for both  https://default-donations.vercel.app/

and 

https://default-donations.vercel.app/donate/how-many-photos-is-too-many-photos

I get this 504 error

![Screen Shot 2022-11-01 at 4 46 20 PM](https://user-images.githubusercontent.com/14945613/199356302-af5dca6c-f16a-4c15-8aed-7c9ece17a1fb.png)

**Comment by CarlosQ96** [comment]
@GriffGreen restarted backend and it's up again! you can keep testing!

**Comment by RamRamez** [comment]
> Wow you are working late Ramin!
> 
> I went to test but can't seem to get the vercel page to load
> 
> for both https://default-donations.vercel.app/
> 
> and
> 
> https://default-donations.vercel.app/donate/how-many-photos-is-too-many-photos
> 
> I get this 504 error
> 
> ![Screen Shot 2022-11-01 at 4 46 20 PM](https://user-images.githubusercontent.com/14945613/199356302-af5dca6c-f16a-4c15-8aed-7c9ece17a1fb.png)

There was a problem with the server, @CarlosQ96 fixed it. Please test again.

**Comment by maryjaf** [comment]
> > ![image.png](https://camo.githubusercontent.com/e3aeefeb30a002a489a0eb19376edb46d9d51a7d6a9e82f6d09f83e6b76aef2d/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3632666537613862353763316131373730353735656134302f38373531353264352d613638342d346635382d623539652d366165316630303435366364) @RamRamez plz check this In fact, it is only a display problem and we do not show 0 GIV, but when we tap on DONATE , only one transaction is shown
> 
> I fixed it. Please check again.

I can't test it, 504: GATEWAY_TIMEOUT

**Comment by maryjaf** [comment]
Plz check this, i can't delete last digit in box for edit amount @RamRamez 
<img width="1435" alt="image" src="https://user-images.githubusercontent.com/111529185/199472239-834b059c-a38d-4014-b7be-6c6cfe91efaf.png">

**Comment by maryjaf** [comment]
<img width="1399" alt="image" src="https://user-images.githubusercontent.com/111529185/199474312-8bb38361-f33f-408b-9f84-b2f2ba590c8f.png">
And in this scenario, this message "Amount is too small"is displayed when the giving amount is less equal 0.0000001 and "Donating to How many photos is t..." value is 0 but "Donating 5% to Giveth" value is  0.000001

**Comment by RamRamez** [comment]
> <img alt="image" width="1434" src="https://user-images.githubusercontent.com/111529185/199475985-becac514-8b35-41c3-b084-0d3b0d7ff97d.png">
> 
> And in this scenario, this message "Amount is too small"is displayed when the giving amount is less equal 0.0000001 and "Donating to How many photos is t..." value is 0 but "Donating 5% to Giveth" value is 0.000001

The min amount for donation is 0.000001 and donate button is disabled

**Comment by RamRamez** [comment]
@maryjaf You raised to bugs, I fixed them both. You can continue testing.

**Comment by maryjaf** [comment]
> @maryjaf You raised two bugs, I fixed them both. You can continue testing.

Thanks @RamRamez 
I tested and it passed

**Comment by divine-comedian** [comment]
@MohammadPCh this can be moved to staging

**Comment by MohammadPCh** [comment]
Thanks @divine-comedian 
I messaged him to keep up.

**Comment by maryjaf** [comment]
I
